NSS-Day-Cohort-42/rare-API-media-moguls
# added custom action publish to post view file

#### Changes Made
1. Modified model for post to allow for null values
1. Added custom action to post.py view file called publish which changes the publication_date value
​
#### Steps to Review
1. Checkout this branch locally.
```
git fetch --all
git checkout rp-publish
```
2. Run the application.
```
npm start
```
3. Open a new Terminal tab (⌘T) and navigate to the server directory.
4. Initialize virtual environment, and run the server.
```
pipenv shell
python manage.py runserver
```
5. Test app functionality.
> nuke the database with ```./nuke.sh```
>in postman run a patch request to ```http://localhost:8000/posts/1/publish```
>run a get request to ```http://localhost:8000/posts/1```
>check that publication_date was changed to null
>repeat the same patch and get request again
>check that publication_date was changed to current date

6. View code file.
> Confirm file modifications are present as indicated above.
> Confirm no unused code or extraneous comments exist.
